### PR TITLE
Answer:56. forms and signal

### PR DESCRIPTION
**Pull request by BillyOkoth**
feat: solution of reactive forms and signals
1. fetch the value from route params
2. convert the observable to signal
3. set the value to the form control

### CODE REVIEW

File: apps/signal/56-forms-and-signal/src/app/checkout.component.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ import { products } from './products';
     <h2 class="mb-1 w-full bg-gray-400 p-2 text-white">Checkout</h2>
     <button
       routerLink="/order"
+      [queryParams]="{ quantity: quantity() }"
```
Comment (tomalaforge) [inline apps/signal/56-forms-and-signal/src/app/checkout.component.ts:18]: why do you need this one?

File: apps/signal/56-forms-and-signal/src/app/order.component.ts [hunk 2]
```diff
@@ -55,11 +57,18 @@ import { products } from './products';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export default class OrderComponent {
+  private route = inject(ActivatedRoute);
+
+  query$ = this.route.queryParams.pipe(map((q) => q['quantity'] as number));
+  productId = input('1');
```
Comment (tomalaforge) [inline apps/signal/56-forms-and-signal/src/app/order.component.ts:63]: you can use input.required here,

File: apps/signal/56-forms-and-signal/src/app/order.component.ts [hunk 3]
```diff
@@ -55,11 +57,18 @@ import { products } from './products';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export default class OrderComponent {
+  private route = inject(ActivatedRoute);
+
+  query$ = this.route.queryParams.pipe(map((q) => q['quantity'] as number));
```
Comment (tomalaforge) [inline apps/signal/56-forms-and-signal/src/app/order.component.ts:62]: you can use the snapshot version since you only want it when the component loads

File: package.json [hunk 4]
```diff
@@ -34,6 +34,7 @@
     "@nx/angular": "19.1.0",
     "@swc/helpers": "0.5.3",
     "@tanstack/angular-query-experimental": "^5.12.1",
+    "node": "^18.18.2",
```
Comment (tomalaforge) [inline package.json:37]: why did you need to add node ?

File: apps/signal/56-forms-and-signal/src/app/checkout.component.ts (hunk 1 above)
Comment (BillyOkoth) [reply apps/signal/56-forms-and-signal/src/app/checkout.component.ts:18]: am passing the quantity data as part of the url in order to retrieve it on the order.component.ts  when we use the ActivatedRoute

File: package.json (hunk 4 above)
Comment (BillyOkoth) [reply package.json:37]: I forgot to remove it since i had accidentally installed it
